receiver/hostmetricsreceiver/internal/scraper/processscraper/process.go

```diff
@@ -109,13 +116,51 @@ func (p *gopsProcessHandles) Len() int {
 	return len(p.handles)
 }
 
+type wrappedProcessHandle struct {
+	*process.Process
+}
+
+func (p wrappedProcessHandle) CgroupWithContext(ctx context.Context) (string, error) {
+	pid := p.Process.Pid
+	statPath := getEnvWithContext(ctx, string(common.HostProcEnvKey), "/proc", strconv.Itoa(int(pid)), "cgroup")
+	contents, err := os.ReadFile(statPath)
+	if err != nil {
+		return "", err
+	}
+
+	return strings.TrimSuffix(string(contents), "\n"), nil
+}
+
+// copied from gopsutil:
```

Comment:
This is not ideal.

I understand that `gopsutil` rejected the proposal. The reason [apparently is](https://github.com/shirou/gopsutil/discussions/1553) that the maintainer of gopsutil wants to introduce a new architecture for platform-specific features in a new major version of the library - v4. Looking at the discussion, the maintainer is a single person working on the library on their weekends. We rely on the library heavily in this component (process scraper), so perhaps it would be nice to contribute to gopsutil, help ship v4 and benefit both projects as a result.

This is not to block this change from merging, but ideally this should be a temporary workaround until gopsutil adds this functionality (with or without our help).

Reply:
Since `gopsutil` is under a different license, should this code copied from `gopsutil` be moved into another file with their license notifier on top? It is BSD licensed with this clause:
```
 * Redistributions of source code must retain the above copyright notice, this
   list of conditions and the following disclaimer.
```

Reply:
@astencel-sumo sure thing, please feel free to offer your help

@braydonk good point, I will take a look

Reply:
Added a NOTICE file with mention of copyright and inclusion of gopsutil code. Note I originated the code to gopsutil, so I could make the argument I can lend copyright usage to both projects.